ui/public/src/main.js

```diff
@@ -24,13 +41,41 @@ export default async function main() {
         break;
       }
       case 'walletNeedDappApproval': {
-        // TODO: toast with message: "Please approve the FungibleFaucet dapp in your wallet"
+        approveDappDialog.open();
         break;
       }
       case 'walletURL': {
         // TODO: handle appropriately
         break;
       }
+      case 'walletUpdatePurses': {
+        // TODO: handle appropriately
+        break;
+      }
+      case 'walletSuggestIssuerResponse': {
+        // TODO: handle appropriately
+        break;
+      }
+      case 'walletSuggestInstallationResponse': {
+        // TODO: handle appropriately
+        break;
+      }
+      case 'walletSuggestInstanceResponse': {
+        // TODO: handle appropriately
+        break;
+      }
+      case 'walletOfferAdded': {
+        // TODO: handle appropriately
```

Comment:
This is where the "approve the offer" toast can appear.

Reply:
Done!